src/main/java/picard/analysis/CollectRnaSeqMetrics.java

```diff
@@ -137,6 +137,11 @@ public static void main(final String[] argv) {
     @Override
     protected void setup(final SAMFileHeader header, final File samFile) {
 
+        // No ribosomal intervals and rRNA fragment percentage = 0
+        if ( (RIBOSOMAL_INTERVALS == null || RIBOSOMAL_INTERVALS.length() == 0) && RRNA_FRAGMENT_PERCENTAGE == 0 ) {
+            throw new PicardException("Must use a RIBOSOMAL_INTERVALS file if RRNA_FRAGMENT_PERCENTAGE = 0.0");
+        }
+
```

Comment:
1. I'd override `customCommandLineValidation` and move this there:
```
@Override
protected String[] customCommandLineValidation() {
  // No ribosomal intervals and rRNA fragment percentage = 0
  if ( (RIBOSOMAL_INTERVALS == null || RIBOSOMAL_INTERVALS.length() == 0) && RRNA_FRAGMENT_PERCENTAGE == 0 ) {
    return new String[]{"Must use a RIBOSOMAL_INTERVALS file if RRNA_FRAGMENT_PERCENTAGE = 0.0"};
  }
  return super.customCommandLineValidation();
}
```
2. I think that you can remove the empty file check here, since we could have a non-empty interval list (just the sequence dictionary) but no intervals, which brings me to...

3. I think if someone goes to the trouble if providing a ribosomal intervals file and there are none read in, we should at least log a warning if not return an error.  This check could be in `makeOverlapDetector`.